handshake.go

```diff
@@ -593,19 +604,23 @@ func (cfg *Config) renewDynamicCertificate(ctx context.Context, hello *tls.Clien
 	renewAndReload := func(ctx context.Context, cancel context.CancelFunc) (Certificate, error) {
 		defer cancel()
 
-		err = cfg.RenewCertAsync(ctx, name, false)
-		if err == nil {
-			// even though the recursive nature of the dynamic cert loading
-			// would just call this function anyway, we do it here to
-			// make the replacement as atomic as possible.
-			newCert, err := cfg.CacheManagedCertificate(name)
-			if err != nil {
-				if log != nil {
-					log.Error("loading renewed certificate", zap.String("server_name", name), zap.Error(err))
+		if revoked {
+			return cfg.forceRenew(ctx, log, currentCert, revokedOCSP)
+		} else {
```

Comment:
Since the `if` branch returns, I don't think you need the `else`, it can be unwrapped.

Reply:
Thanks. I _knew_ something looked funny about that... been a long day.